src/ui/public/courier/data_source/_abstract.js

```diff
@@ -389,6 +391,8 @@ export default function SourceAbstractFactory(Private, Promise, PromiseEmitter)
             recurse(agg.aggs || agg.aggregations);
           });
         }(flatState.body.aggs || flatState.body.aggregations));
+
+        highlightBody(flatState.body);
```

Comment:
I would definitely vote for not adding more magic to the data source. This piece of code has cost me so much time when writing the log context view before giving up on it. The more we add the more we restrict its reuse.